lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,495 @@
+      'and the up arrow key is pressed', async function() {
+      const user = userEvent.setup(),
+          { getByRole, getAllByRole } = quickRender({
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+          inputElement = getByRole('combobox'),
+          firstOptBtn = getAllByRole('option')[0],
+          secondOptBtn = getAllByRole('option')[1];
+
+      inputElement.focus();
+      await user.keyboard('[ArrowUp]');
+      expect(secondOptBtn).toHaveAttribute('aria-selected', 'true');
+      expect(inputElement).toHaveAttribute('aria-activedescendant', secondOptBtn.id);
+      await user.keyboard('[ArrowUp]');
+      expect(secondOptBtn).toHaveAttribute('aria-selected', 'false');
+      expect(firstOptBtn).toHaveAttribute('aria-selected', 'true');
+      expect(inputElement).toHaveAttribute('aria-activedescendant', firstOptBtn.id);
+    });
+
+    it('should move visual focus to the last option if current focus is on the first option' +
+      'and the up arrow key is pressed', async function() {
+      const user = userEvent.setup(),
+          { getByRole, getAllByRole } = quickRender({
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }] }),
+          inputElement = getByRole('combobox'),
+          firstOptBtn = getAllByRole('option')[0],
+          secondOptBtn = getAllByRole('option')[1];
+
+      inputElement.focus();
+      await user.keyboard('[ArrowDown]');
+      expect(firstOptBtn).toHaveAttribute('aria-selected', 'true');
+      expect(inputElement).toHaveAttribute('aria-activedescendant', firstOptBtn.id);
+      await user.keyboard('[ArrowUp]');
+      expect(firstOptBtn).toHaveAttribute('aria-selected', 'false');
+      expect(secondOptBtn).toHaveAttribute('aria-selected', 'true');
+      expect(inputElement).toHaveAttribute('aria-activedescendant', secondOptBtn.id);
+    });
+
+    it('sets the value to be backspaced text when backspace is pressed', async function() {
+      const user = userEvent.setup(),
+          onChange = jest.fn(),
+          { getByRole } = quickRender({ value: 'foo', onChange }),
+          inputElement = getByRole('combobox') as HTMLInputElement;
+
+      inputElement.focus();
+      inputElement.setSelectionRange(1, 2);
+      await user.keyboard('[Backspace]');
+      expect(onChange).toHaveBeenCalledWith('fo');
+    });
+
+    it('should clear input text if dropdown is close when Escape key is pressed', async function() {
+      const user = userEvent.setup(),
+          onChange = jest.fn(),
+          { getByRole } = quickRender({
+            value: 'a',
+            matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }],
+            onChange}),
+          inputElement = getByRole('combobox');
+
+      inputElement.focus();
+      await user.keyboard('[Escape]');
+      expect(onChange).toBeCalledWith('');
+    });
+  });
```

Comment:
Needs tests for the validation-related props.

Reply:
> Needs tests for the validation-related props.

The most thorough existing example of RTL-based validation tests would in `NxFormSelect.test.tsx` on the `FEATURE_RSC-976_required_form_fields` branch, so you might start by copying and adapting those.

Reply:
Updated here 25714fe4c574445a1b0416abb60d789c45efb98f.